src/frequenz/sdk/actor/power_distributing/_component_status/_pv_inverter_status_tracker.py

```diff
@@ -0,0 +1,205 @@
+import logging
+from datetime import datetime, timedelta, timezone
+
+# Component state for inverters and batteries is not wrapped by the
+# microgrid client currently, so it needs to be imported directly from
+# the api repo.
+# pylint: disable=no-name-in-module
+from frequenz.api.microgrid.inverter_pb2 import (
+    ComponentState as PbInverterComponentState,
+)
+
+# pylint: enable=no-name-in-module
+from frequenz.channels import Receiver, Sender, select, selected_from
+from frequenz.channels.timer import SkipMissedAndDrift, Timer
+from frequenz.client.microgrid import InverterData
+from typing_extensions import override
+
+from ....microgrid import connection_manager
+from ..._background_service import BackgroundService
+from ._blocking_status import BlockingStatus
+from ._component_status import (
+    ComponentStatus,
+    ComponentStatusEnum,
+    ComponentStatusTracker,
+    SetPowerResult,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class PVInverterStatusTracker(ComponentStatusTracker, BackgroundService):
+    """Status tracker for PV inverters.
+
+    It reports a PV inverter as `WORKING` or `NOT_WORKING` based on
+    the status in the received component data from the microgrid API.
+    When no data is received for a specific duration, the component is
+    marked as `NOT_WORKING`.
+
+    If it receives a power assignment failure from the PowerDistributor,
+    when the component is expected to be `WORKING`, it is marked as
+    `UNCERTAIN` for a specific interval, before being marked `WORKING`
+    again.
+    """
+
+    @override
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        component_id: int,
+        max_data_age: timedelta,
+        max_blocking_duration: timedelta,
+        status_sender: Sender[ComponentStatus],
+        set_power_result_receiver: Receiver[SetPowerResult],
+    ) -> None:
+        """Initialize this instance.
+
+        Args:
+            component_id: ID of the PV inverter to monitor the status of.
+            max_data_age: max duration to wait for, before marking a component as
+                NOT_WORKING, unless new data arrives.
+            max_blocking_duration: max duration to wait for, before marking a component
+                as BLOCKING, unless new data arrives.
+            status_sender: Sender to send the status of the PV inverter.
+            set_power_result_receiver: Receiver for the power assignment result.
+        """
+        self._component_id = component_id
+        self._max_data_age = max_data_age
+        self._status_sender = status_sender
+        self._set_power_result_receiver = set_power_result_receiver
+
+        self._last_status = ComponentStatusEnum.NOT_WORKING
+        self._blocking_status = BlockingStatus(
+            min_duration=timedelta(seconds=1.0),
+            max_duration=max_blocking_duration,
+        )
+
+        BackgroundService.__init__(
+            self, name=f"PVInverterStatusTracker({component_id})"
+        )
```

Comment:
Normally the parent constructor should be called first (otherwise you can be using wrongly initialized parent data), if there is a specific reason to call it last I would add a comment.
```suggestion
        BackgroundService.__init__(
            self, name=f"PVInverterStatusTracker({component_id})"
        )
        self._component_id = component_id
        self._max_data_age = max_data_age
        self._status_sender = status_sender
        self._set_power_result_receiver = set_power_result_receiver

        self._last_status = ComponentStatusEnum.NOT_WORKING
        self._blocking_status = BlockingStatus(
            min_duration=timedelta(seconds=1.0),
            max_duration=max_blocking_duration,
        )
```